libs/permission-checker/tests/Check/OAuth/CanModifyAuthorizationTest.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\PermissionChecker\Tests\Check\OAuth;
+
+use Keboola\PermissionChecker\BranchType;
+use Keboola\PermissionChecker\Check\OAuth\CanModifyAuthorization;
+use Keboola\PermissionChecker\Exception\PermissionDeniedException;
+use Keboola\PermissionChecker\StorageApiToken;
+use PHPUnit\Framework\TestCase;
+
+class CanModifyAuthorizationTest extends TestCase
+{
+    public static function provideValidPermissionsCheckData(): iterable
+    {
+        yield 'regular token on regular project default branch' => [
+            'branchType' => BranchType::DEFAULT,
+            'token' => new StorageApiToken(role: null),
```

Comment:
neznamená to náhodou, že to může udělat readonly user svým autosave tokenem?

Reply:
:shrug: jak vypada auto-save token?

Reply:
no nemá roli

Reply:
imho se to bude muset pověsit na nějaky canManageBuckets nebo nějaký takový psycho řešení. On je to vlastně podobnej problém jako tohle https://keboolaglobal.slack.com/archives/C05BK5V8N1Z/p1692258992951869

Reply:
eh, uff.. tim padem bude potreba asi zrevidovat vsechny ty checkery :disappointed: